apis/placement/v1beta1/overridesnapshot_types.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope="Cluster",shortName=rss,categories={fleet,fleet-placement}
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=`.metadata.generation`,name="Gen",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterResourceOverrideSnapshot is used to store a snapshot of ClusterResourceOverride.
+// Its spec is immutable.
+// We assign an ever-increasing index for snapshots.
+// The naming convention of a ClusterResourceOverrideSnapshot is {ClusterResourceOverride}-{resourceIndex}.
+// resourceIndex will begin with 0.
+// Each snapshot MUST have the following labels:
+//   - `ClusterResourceOverrideTrackingLabel` which points to its owner ClusterResourceOverride.
+//   - `IsLatestSnapshotLabel` which indicates whether the snapshot is the latest one.
+type ClusterResourceOverrideSnapshot struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of ClusterResourceOverrideSnapshotSpec.
+	// +required
+	Spec ClusterResourceOverrideSnapshotSpec `json:"spec"`
+}
+
+type ClusterResourceOverrideSnapshotSpec struct {
+	OverrideName string `json:"overrideName"`
```

Comment:
comment on this field? Is this different from the label value "ClusterResourceOverrideTrackingLabel"?